Source/WebCore/Modules/webdatabase/Database.cpp

```diff
@@ -92,19 +92,13 @@ namespace WebCore {
 // (and un-returned) databases instances.
 
 static const char versionKey[] = "WebKitDatabaseVersionKey";
-static const char unqualifiedInfoTableName[] = "__WebKitDatabaseInfoTable__";
+static constexpr auto unqualifiedInfoTableName = "__WebKitDatabaseInfoTable__"_s;
 const unsigned long long quotaIncreaseSize = 5 * 1024 * 1024;
 
-static const char* fullyQualifiedInfoTableName()
+static const String& fullyQualifiedInfoTableName()
 {
-    static const char qualifier[] = "main.";
-    static char qualifiedName[sizeof(qualifier) + sizeof(unqualifiedInfoTableName) - 1];
-
-    static std::once_flag onceFlag;
-    std::call_once(onceFlag, [] {
-        snprintf(qualifiedName, sizeof(qualifiedName), "%s%s", qualifier, unqualifiedInfoTableName);
-    });
-
+    static constexpr auto qualifier = "main."_s;
+    static NeverDestroyed<String> qualifiedName = makeString(qualifier, unqualifiedInfoTableName);
```

Comment:
Was the `std::call_once` really pointless here?

Reply:
I am not 100% confident so I'll restore it and use LazyNeverDestroyed.

Reply:
The old code had a global array of characters. Not a global `String` used across threads. Made a new string each time it was called. I don’t think `LazyNeverDestroyed` can replicate that level of thread safety.

Reply:
I was thinking of using a StaticStringImpl inside the String, those are thread safe and made for usage like this AFAIK.